src/layers/mpls.rs

```diff
@@ -0,0 +1,151 @@
+//! Multi Label Protocol Switching
+
+use std::convert::TryInto;
+
+use serde::Serialize;
+
+use crate::errors::Error;
+use crate::layers::ethernet;
+use crate::types::{ETHERTYPE_MPLS_MULTICAST, ETHERTYPE_MPLS_UNICAST};
+use crate::Layer;
+
+use super::ipv4::IPv4;
+
+/// Default Header Length for MPLS Packets
+pub const MPLS_HDR_LEN: usize = 4_usize;
```

Comment:
Currently there's a bit of inconsistency in the way we use these variable names: eg. sometimes we use `FOO_HDR_LEN`, and sometimes `FOO_HEADER_LENGTH`. I am merging a branch for fixing these changes. I believe, you will have to use and then use the `MPLS_HEADER_LENGTH` as the `const` name.